DOCS-688: align configuration instructions across updated components

sguequierre opened the pull request:
* add config tab sub-instruction & create component instructions in Config Builder tab
* change images to markdown across docs
* remove Configuration headings 
* Combine attributes tables & standardize types: JSON only recognizes "number", but "int" is a subset of number-- so I changed int* to int and float* to number

npentrel commented on docs/components/motor/28byj-48.md:
```diff
@@ -17,17 +17,14 @@ Viam also supports:
 
-## Configuration
-
-To configure a `28byj48` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
This is needed

sguequierre replied:
Added back on top!

npentrel commented on docs/components/motor/28byj-48.md:
```diff
@@ -17,17 +17,14 @@ Viam also supports:
-To configure a `28byj48` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`28byj48`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
I think we should also keep that

sguequierre replied:
Added 22 & 25 back in on top and attributes, respectively-- 24 is covered

npentrel commented on docs/components/motor/dmc4000.md:
```diff
@@ -15,17 +15,16 @@ When using it, you do not need to configure a [board component](/components/boar
-## Configuration
-
-Assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`DMC4000`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
I think adding this line near the attributes table is helpful

sguequierre replied:
Kk, added 21 back in under attributes

npentrel commented on docs/components/motor/gpio/_index.md:
```diff
@@ -12,19 +12,16 @@ The `gpio` model supports [DC motors](https://en.wikipedia.org/wiki/DC_motor) (b
 
-## Configuration
-
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Need to cover this

sguequierre replied:
Added!

npentrel commented on docs/components/motor/gpio/_index.md:
```diff
@@ -12,19 +12,16 @@ The `gpio` model supports [DC motors](https://en.wikipedia.org/wiki/DC_motor) (b
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`gpio`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Near the attributes table would be good

npentrel commented on docs/components/motor/gpiostepper.md:
```diff
@@ -13,19 +13,16 @@ The `gpiostepper` model of the motor component supports bipolar [stepper motors]
 
-## Configuration
-
-To configure a `gpiostepper` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Needed

sguequierre replied:
Added!

npentrel commented on docs/components/motor/gpiostepper.md:
```diff
@@ -13,19 +13,16 @@ The `gpiostepper` model of the motor component supports bipolar [stepper motors]
-To configure a `gpiostepper` motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`gpiostepper`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Maybe useful near attributes table

npentrel commented on docs/components/motor/tmc5072.md:
```diff
@@ -11,19 +11,18 @@ This model supports stepper motors controlled by the [TMC5072 chip](https://www.
 
-## Configuration
-
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
```
Needed

sguequierre replied:
Added!

npentrel commented on docs/components/motor/tmc5072.md:
```diff
@@ -11,19 +11,18 @@ This model supports stepper motors controlled by the [TMC5072 chip](https://www.
-To configure a DC motor as a component of your robot, first configure the [board](/components/board/) to which the motor driver is wired.
-
-Then assign your motor a `name` (to identify the motor), `type` (`motor`) and `model` (`TMC5072`), and fill in the [attributes](#required-attributes) that apply to your particular hardware.
-Refer to your motor and motor driver data sheets for specifics.
```
Maybe useful near attributes table

npentrel commented:
We sometimes say "The following attributes are available for ~:" before the tables and sometimes not. We can clean that up in a separate PR though if you want

sguequierre commented:
> We sometimes say "The following attributes are available for ~:" before the tables and sometimes not. We can clean that up in a separate PR though if you want

Tried to cover this across motor when addressing feedback there, might need to make a separate pr/full ticket for everything

sguequierre commented on docs/components/motor/dmc4000.md:
```diff
@@ -83,24 +82,21 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
+| `home_rpm` | int | **Required** | Set speed in revolutions per minute (RPM) that the motor will turn when executing a Home() command (using  DoCommand()). |
+| `dir_flip` | bool | Optional | Flip the direction of the signal sent if there is a DIR pin. |
+| `max_rpm` | int | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
+| `max_acceleration_rpm_per_sec` | int | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```
```suggestion
| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |

```

sguequierre commented on docs/components/motor/dmc4000.md:
```diff
@@ -83,24 +82,21 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
+| `controller_axis` | string | **Required** | Physical port label (A-H); select which "axis" the motor is wired to on the controller. |
+| `home_rpm` | int | **Required** | Set speed in revolutions per minute (RPM) that the motor will turn when executing a Home() command (using  DoCommand()). |
+| `dir_flip` | bool | Optional | Flip the direction of the signal sent if there is a DIR pin. |
+| `max_rpm` | int | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
```
```suggestion
| `max_rpm` | number | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |

```

sguequierre commented on docs/components/motor/dmc4000.md:
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
-`serial_path` | string | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path.
-`controller_axis` | string | Physical port label (A-H); select which "axis" the motor is wired to on the controller.
-`home_rpm` | integer | Speed in revolutions per minute that the motor will turn when executing a Home() command (using  DoCommand()).
+| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```
```suggestion
| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```

sguequierre replied:
``` suggestion
```

sguequierre commented on docs/components/motor/dmc4000.md:
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
+| `controller_axis` | string | **Required** | Physical port label (A-H); select which "axis" the motor is wired to on the controller. |
+| `home_rpm` | int | **Required** | Set speed in revolutions per minute (RPM) that the motor will turn when executing a Home() command (using  DoCommand()). |
+| `dir_flip` | bool | Optional | Flip the direction of the signal sent if there is a DIR pin. |
+| `max_rpm` | number | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
```
```suggestion
| `max_rpm` | number | Optional | Set a limit on maximum revolutions per minute that the motor can be run at. |
| `max_acceleration_rpm_per_sec` | number | Optional | Set the maximum revolutions per minute (RPM) per second acceleration limit. |
```

sguequierre commented on docs/components/motor/dmc4000.md:
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
-`serial_path` | string | Path to <file>/dev/ttyXXXX</file> file. You can find the `XXXX` by running `dmesg` on the computer plugged into the DMC-40x0. If you leave this empty, Viam will attempt to automatically detect the path.
-`controller_axis` | string | Physical port label (A-H); select which "axis" the motor is wired to on the controller.
-`home_rpm` | integer | Speed in revolutions per minute that the motor will turn when executing a Home() command (using  DoCommand()).
 
```
```suggestion
```

sguequierre commented on docs/components/motor/dmc4000.md:
```diff
@@ -83,24 +82,23 @@ Example configuration for a stepper motor controlled by a DMC-40x0 motion contro
-`home_rpm` | integer | Speed in revolutions per minute that the motor will turn when executing a Home() command (using  DoCommand()).
 
-### Optional Attributes
 
```
```suggestion
```
